gbpservice/nfp/configurator/drivers/firewall/paloalto/paloalto_fw_driver.py

```diff
@@ -0,0 +1,875 @@
+                    raise pan.config.PanConfigError(err)
+                except Exception as err:
+                    self._print_exception('UnexpectedError', err,
+                                          const.DELETE_SECURITY_SERVICE, resp)
+                    raise Exception(err)
+
+                msg = "DELETED security service '%s' to Service VM"\
+                                                            % service_name
+                if self.analyze_response(msg, resp,
+                                         const.DELETE_SECURITY_SERVICE)\
+                                            == const.STATUS_ERROR:
+                    return const.STATUS_ERROR
+            else:
+                application_name = 'ping-' + curr_rule['protocol']        
+                try:
+                    resp = self.delete_security_application(vm_mgmt_ip,
+                                                            application_name)
+                except pan.xapi.PanXapiError as err:
+                    self._print_exception('PanXapiError', err,
+                                          const.DELETE_APPLICATION_SERVICE)
+                    raise pan.xapi.PanXapiError(err)
+                except pan.config.PanConfigError as err:
+                    self._print_exception('PanConfigError', err,
+                                          const.DELETE_APPLICATION_SERVICE)
+                    raise pan.config.PanConfigError(err)
+                except Exception as err:
+                    self._print_exception('UnexpectedError', err,
+                                      const.DELETE_APPLICATION_SERVICE, resp)
+                    raise Exception(err)
+
+                msg = "DELETED security application '%s' to Service VM"\
+                                                            % application_name
+                if self.analyze_response(msg, resp,
+                                        const.DELETE_SECURITY_APPLICATION)\
+                                         == const.STATUS_ERROR:
+                    return const.STATUS_ERROR
+
+        # commit
+        try:
+            resp = self.commit(hostname=vm_mgmt_ip)
+        except pan.xapi.PanXapiError as err:
+            self._print_exception('PanXapiError', err, const.COMMIT)
+            raise pan.xapi.PanXapiError(err)
+        except pan.config.PanConfigError as err:
+            self._print_exception('PanConfigError', err, const.COMMIT)
+            raise pan.config.PanConfigError(err)
+        except Exception as err:
+            self._print_exception('UnexpectedError', err, const.COMMIT, resp)
+            raise Exception(err)
+  
+        if self.analyze_response("COMMITED the configuration to Service VM",
+                                 resp, const.COMMIT) == const.STATUS_ERROR:
+            return const.STATUS_ERROR
+
+        return const.STATUS_DELETED
+
+    def create_security_rule_element(self, rule_name,
```

Comment:
Lets prefix all the internal functions (with in the file) with "_". Only the crud of firewall and configure/clear interface/routes will be the APIs of the driver.